### ISSUE TITLE
Notebooks hierarchy in MF repositories

### ISSUE THREAD

**Issue by dvorka** [dvorka/mindforger#745]
Introduce view and notebook metadata allowing to create hierarchy of notebooks in a MF repository (one repo - one position in hierarchy). This feature requires new UI view and new notebook metadata (parent + depth).

Remind O hierarchy in MR.

**Comment by StoltHD** [comment]
Yes please, that would be so helpfull and I would be able to fully extend my personal research to mindforger...
And please let it be without limits in depth or number of "Nodes" (Notebooks)...
And make it possible to add all the notebooks in a "tree" to a printout/export as paragraphs/chapters...

You have a great software here, with even greater potential...

**Comment by dvorka** [comment]
Thank you for the feedback @StoltHD! I just increased priority of this issue as several users asked for this.

**Comment by casperl** [comment]
> Introduce view and notebook metadata allowing to create hierarchy of notebooks in a MF repository (one repo - one position in hierarchy). 

I have a question, would it not be more sensible to implement this hierarchy in the file system as subdirectories instead of files with meta tags?  Not only might that make the programming task easier, but it will have the side effect that all files related to any specific notebook in a lower hierarchy will be in the same place.  Ideally such hierarchies should be as many theoretical levels deep as the operating system will allow.

As far as I can ascertain, and I am still a MF newbie, all files are stored in a flat file structure in the my-repository/memory directory.

In a closely related issue, I created a symlink to vimwiki containing files created in VimWiki into my-repsitory/memory/vimwiki as a hiarachical set of subdirectories, but unless I am missing something, MF did not specifically discover and index those files.  (For the record, they are all MarkDown files since VimWIki supports either wiki or markdown formats.)   If so, I have a big decision ahead to decide whether to fully convert to MF since the way I set up many thousands of VimWiki files was one file + support files per directory in a vast hierarchical system.   The reason VimWIki is not as functional as can be, is that increasingly I have to rely on lengthy ack and fzf searches to manage that knowledgebase from within Vim.

MindForger hits just the right spot of visual overview, great power, yet greater simplicity for me.
